### PR TITLE
LinkFix: azure-docs-cli (2022-03)

### PR DESCRIPTION
**Pull request by cmcclister**
This PR is the result of running a Link Repair script on the included content. This script is primarily being run to clean up links to function correctly in Air Gapped Clouds (AGC). Here's a breakdown of what the script that generated the changes in this PR fixes: 

 

- docs.microsoft.com Fully Qualified Domain Name (FQDN) Links to either Relative (if in same Repo/DocSet/ContentSet) and Site Relative if not 

- azure.microsoft.com/documentation/articles that reflect to Docs to either Relative (if in same Repo/DocSet/ContentSet) and Site Relative if not 

- All known flavors of MSDN/TechNet domains (we know of 8 or 9 now) that reflect to Docs to either Relative (if in same Repo/DocSet/ContentSet) and Site Relative if not 

  - This also includes cleanup of the appended query string for MSDN (redirectedfrom=MSDN). 

- Fixes articles that are redirected in `.openpublishing.redirection.json` and updates the link to the current final location (this has a lot of customer usability fixes). 

 

Please note the [Contributor Guide: Links](https://review.docs.microsoft.com/en-us/help/contribute/links-how-to?branch=master) has been updated with the following link-type preference order: 

 

``` 

By order of preference, links hosted on Docs should be Relative if they are in the same repository and docset. If they are in a different docset, even if in the same repository, they should be Site Relative. Links to content hosted on Docs shouldn't use a complete URL, otherwise known as Fully Qualified Domain Names (FQDN). Using a complete URL from Docs to other content on Docs will cause that link to be non-functional in air-gap environments. 

```

### CODE REVIEW

**Comment by cmcclister** [comment]
#sign-off

File: docs-ref-conceptual/azure-cli-extensions-list.md [hunk 1]
```diff
@@ -36,12 +36,12 @@ You will be prompted to install an extension on first use.
...
 |[automation](https://github.com/Azure/azure-cli-extensions/tree/master/src/automation) | 2.13.0 | Microsoft Azure Command-Line Tools AutomationClient Extension | Experimental | [0.1.1](https://github.com/Azure/azure-cli-extensions/tree/master/src/automation/HISTORY.rst) |
 |[azure-batch-cli-extensions](https://github.com/Azure/azure-batch-cli-extensions) | 2.0.74 | Additional commands for working with Azure Batch service | GA | [6.0.0](https://github.com/Azure/azure-batch-cli-extensions) |
-|[azure-cli-ml](https://docs.microsoft.com/python/api/overview/azure/ml/?view=azure-ml-py) | 2.3.1 | Microsoft Azure Command-Line Tools AzureML Command Module | GA | [1.37.0](https://docs.microsoft.com/python/api/overview/azure/ml/?view=azure-ml-py/HISTORY.rst) |
+|[azure-cli-ml](/python/api/overview/azure/ml/?view=azure-ml-py) | 2.3.1 | Microsoft Azure Command-Line Tools AzureML Command Module | GA | [1.37.0](/python/api/overview/azure/ml/?view=azure-ml-py%2fHISTORY.rst) |
```
Comment (chasewilson) [inline docs-ref-conceptual/azure-cli-extensions-list.md:44]: I believe the version link is supposed to go to a github repo right?

**Review by chasewilson — APPROVED** [review]
Looks good with my one comment as an exception. If we can get that fixed before merging I'm good. :)

File: docs-ref-conceptual/azure-cli-extensions-list.md (hunk 1 above)
Comment (dbradish-microsoft) [reply docs-ref-conceptual/azure-cli-extensions-list.md:44]: @cmcclister, the `azure-cli-extensions-list.md` file is autogenerated so any update needs to be in the source code.  The CLI extension data also originates in multiple repos as it is owned by different Azure service teams.  For a list of extensions owned by the Azure CLI engineering team, see [Modules owned by CLI team](https://github.com/Azure/azure-cli/blob/dev/doc/modules_owned_by_cli_team.md). 

[azure-cli-extensions](https://github.com/Azure/azure-cli-extensions) is our team's extension repo.  @chasewilson can probably assist you with finding all other teams / source code repos.

**Pull request merged**